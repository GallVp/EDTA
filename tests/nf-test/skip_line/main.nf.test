nextflow_pipeline {

    name "Test no LINE skip and no annotation with a small genome"
    script "main.nf"

    test("small genome -- no LINE -- no annotation") {

        when {
            params {
                genome                      = "https://raw.githubusercontent.com/oushujun/EDTA/4f0424b8a3ce2caccdd51b4021c61a536fa9ead9/test/genome.fa"
                anno                        = false
                publish_raw_filtered_libs   = true
                publish_stage1_libs         = true
                outdir                      = "$outputDir"

                line_detection_tool         = 'none'
            }
        }

        then {

            def stable_path = getAllFilesFromDir(
                params.outdir,
                false,
                [
                    'pipeline_info/*.{html,json,txt,yml}',
                    'final/genome/genome.intact.fa',
                    'final/genome/genome.intact.gff3',
                    'final/genome/genome.TElib.fa',
                    'raw/**',
                    'stage1/**'
                ],
                null,
                ['**']
            )

            def stable_name = getAllFilesFromDir(
                params.outdir,
                true,
                [
                    'pipeline_info/*.{html,json,txt,yml}'
                ],
                null,
                ['**']
            )

            def te_lib_count = file("$outputDir/final/genome/genome.TElib.fa").readLines().findAll { it.startsWith('>') }.size()
            def intact_lib_count = file("$outputDir/final/genome/genome.intact.fa").readLines().findAll { it.startsWith('>') }.size()

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    [
                        'successful tasks': workflow.trace.succeeded().size(),
                        'versions': removeNextflowVersion("$outputDir/pipeline_info/software_versions.yml"),
                        'stable paths': stable_path,
                        'stable names': getRelativePath(stable_name, outputDir)
                    ]
                ).match() },
                { assert Math.abs(te_lib_count - 45) <= 5.0 },
                { assert Math.abs(intact_lib_count - 42) <= 5.0 }
            )
        }

    }

}
