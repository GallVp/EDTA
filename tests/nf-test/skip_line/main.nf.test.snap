{
    "small genome -- no LINE -- no annotation": {
        "content": [
            {
                "successful tasks": 21,
                "versions": {
                    "ANNOSINE": {
                        "annosine": "2.0.7"
                    },
                    "CAT_CAT": {
                        "pigz": "2.3.4"
                    },
                    "COMBINE_INTACT_TES": {
                        "perl": "v5.32.1"
                    },
                    "CUSTOM_SHORTENFASTAIDS": {
                        "python": "3.8.13",
                        "biopython": 1.75
                    },
                    "FINAL_FILTER": {
                        "perl": "v5.32.1",
                        "repeatmasker": "4.1.7-p1"
                    },
                    "FORMAT_HELITRONSCANNER_OUT": {
                        "perl": "v5.32.1"
                    },
                    "FORMAT_HELITRONSCANNER_OUT_EXT": {
                        "perl": "v5.32.1"
                    },
                    "HELITRONSCANNER_DRAW": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_DRAW_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "HELITRONSCANNER_SCAN_HEAD": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_HEAD_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "LTRFINDER": {
                        "LTR_FINDER_parallel": "v1.1",
                        "ltr_finder": "v1.07"
                    },
                    "LTRHARVEST": {
                        "LTR_HARVEST_parallel": "v1.1",
                        "genometools": "1.6.5"
                    },
                    "LTRRETRIEVER_LTRRETRIEVER": {
                        "LTR_retriever": "v2.9.9"
                    },
                    "LTR_RETRIEVER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "PROCESS_K": {
                        "perl": "v5.32.1",
                        "repeatmasker": "4.1.7-p1"
                    },
                    "TIRLEARNER": {
                        "TIR-Learner": "v3.0.2"
                    },
                    "TIR_LEARNER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "Workflow": {
                        "oushujun/EDTA": "v3.0.0dev"
                    }
                },
                "stable paths": [
                    "genome.short.ids.tsv:md5,642382addc4beba37088b1ebe09d38cf"
                ],
                "stable names": [
                    "final",
                    "final/genome",
                    "final/genome/genome.TElib.fa",
                    "final/genome/genome.intact.fa",
                    "final/genome/genome.intact.gff3",
                    "final/genome/genome.short.ids.tsv",
                    "pipeline_info",
                    "raw",
                    "raw/filtered",
                    "raw/filtered/helitron",
                    "raw/filtered/helitron/genome",
                    "raw/filtered/helitron/genome/genome.Helitron.intact.anno.list",
                    "raw/filtered/helitron/genome/genome.Helitron.intact.fa",
                    "raw/filtered/ltr",
                    "raw/filtered/ltr/genome",
                    "raw/filtered/ltr/genome/genome.LTR.fa",
                    "raw/filtered/ltr/genome/genome.LTR.intact.anno.list",
                    "raw/filtered/ltr/genome/genome.LTR.intact.fa",
                    "raw/filtered/tir",
                    "raw/filtered/tir/genome",
                    "raw/filtered/tir/genome/genome.TIR.intact.fa",
                    "stage1",
                    "stage1/genome",
                    "stage1/genome/genome.fasta",
                    "stage1/genome/genome.intact.fasta"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-19T14:52:45.33513"
    }
}